### ISSUE TITLE
Project Dates/Hours - not saving correctly

### ISSUE THREAD

**Issue by JJMilmore** [dandahle/Catalyst-AppVetting#294]
The project Dates and Times are not saving correctly in Production.  When updating the project dates under "Project Rodin - Roof" to 9/24 to 9/26, it appears to save.  I can switch tabs (work Items, Cost Summary) and the new dates display correctly when returning to the Info tab.  But after refreshing the screen or returning to the Projects main view, they would change back and/or have different hour start/end times.

**Comment by dandahle** [comment]
Date and time seemed to change on their own when I tested it. Other changes in Planning and Wrap 
![294-Screenshot 2021-07-13 215642](https://user-images.githubusercontent.com/6306417/125564857-0a916508-6d96-469d-a48e-8a1194aa9987.png)
![294-Screenshot 2021-07-13 220733-refresh4](https://user-images.githubusercontent.com/6306417/125564878-327c763e-2c10-4a11-be52-302a0d19e4bd.png)
![294-Screenshot 2021-07-13 215814-chg1](https://user-images.githubusercontent.com/6306417/125564880-833305aa-a2a6-4a96-a7a1-0571fe6e4a47.png)
![294-Screenshot 2021-07-13 220018-chg2](https://user-images.githubusercontent.com/6306417/125564881-4fbcc2f5-6b84-4112-9e6d-10cd65430d1d.png)
![294-Screenshot 2021-07-13 220107-chg3](https://user-images.githubusercontent.com/6306417/125564883-2f366dfe-8d34-4574-a2f4-fd9a34de4ce3.png)
![294-Screenshot 2021-07-13 220159-refresh1](https://user-images.githubusercontent.com/6306417/125564884-5d3691c6-acc2-436c-8b53-1dcce3782724.png)
![294-Screenshot 2021-07-13 220243-refresh2](https://user-images.githubusercontent.com/6306417/125564885-35f28b52-5eac-4cb0-be2e-c843be03ee0f.png)
![294-Screenshot 2021-07-13 220324-refresh3](https://user-images.githubusercontent.com/6306417/125564887-d71d7c28-187a-4b3a-a3ac-ea762be31635.png)
![294-Screenshot 2021-07-13 220635-chg4](https://user-images.githubusercontent.com/6306417/125564888-fa4888d7-4d37-42ad-85e7-43507ada8816.png)
Up seemed to behave as expected.

**Comment by rarafon** [comment]
I correct the project controller, but the issue was that when you wanted to set the date for a certain month, the server would use the previous month.

For the planning and wrap-up tab, I was unable to reproduce the bug. Does this happen for any application?

**Comment by dandahle** [comment]
I pulled your changes into develop-2020-aug and tested them on http://52.35.59.80/. I think you are on the right track, Andy, but there is still something going on with the time. When a change is made it seems to count back 7 hours. I didn't really notice this changing days. Sometimes when I changed a day and refreshed the date changed and sometimes it didn't. Then I noticed the time change. The date changes when the time counts back 7 hours from 6am or earlier. Here are the screen captures where I changed the time, then clicked the page refresh button. This same behavior happens with the Start Date and End Date. 
![294-Screenshot 2021-07-14 093714-start](https://user-images.githubusercontent.com/6306417/125661524-83cd0ee1-5d3e-4a38-b4cb-b5c5d6ffb25f.png)
![294-Screenshot 2021-07-14 093821-chg1](https://user-images.githubusercontent.com/6306417/125661523-ee4f3e0f-12ba-4584-94cc-7b7bc80fe9ad.png)
![294-Screenshot 2021-07-14 093912-refresh1](https://user-images.githubusercontent.com/6306417/125661519-1f2357de-63ad-4ed1-b4a0-908d9085241b.png)
![294-Screenshot 2021-07-14 093957-chg2](https://user-images.githubusercontent.com/6306417/125661518-e09cfeb7-d6cd-4560-a012-cdfea243a9d8.png)
![294-Screenshot 2021-07-14 094035-refresh2](https://user-images.githubusercontent.com/6306417/125661516-ad7fbef3-1a9a-4bf3-8f73-21ad1369e7b6.png)
![294-Screenshot 2021-07-14 094126-chg3](https://user-images.githubusercontent.com/6306417/125661514-007cc403-d11d-4fe0-b355-e88689cc9f6e.png)
![294-Screenshot 2021-07-14 094206-refresh3](https://user-images.githubusercontent.com/6306417/125661512-5cd5503a-8030-49c5-ad0c-cc49970fb86a.png)
![294-Screenshot 2021-07-14 094309-chg4](https://user-images.githubusercontent.com/6306417/125661508-27845d50-8f83-4ab4-9c91-dc872976817b.png)
![294-Screenshot 2021-07-14 094358-refresh4](https://user-images.githubusercontent.com/6306417/125661531-41fea491-bca8-4f39-b74e-1f39dab23410.png)
![294-Screenshot 2021-07-14 094437-chg5](https://user-images.githubusercontent.com/6306417/125661530-563ff948-f405-4930-b000-9c9a45d024f4.png)
![294-Screenshot 2021-07-14 094523-refresh5](https://user-images.githubusercontent.com/6306417/125661528-5ec3e5df-e43b-4eaa-b50d-4a0d4a31ccd4.png)
![294-Screenshot 2021-07-14 094609-chg6](https://user-images.githubusercontent.com/6306417/125661527-4092d972-b3ca-4bc2-95fa-8f2e6aed824e.png)
![294-Screenshot 2021-07-14 094702-refresh6](https://user-images.githubusercontent.com/6306417/125661525-3eb2b3d3-f339-406c-bf2b-86806c73d54c.png)

**Comment by dandahle** [comment]
Also, it was only the date saving that I saw issues with. I changed items in Planning, Partner and WrapUp tabs and everything looked like it saved OK, so once we have the date issue fixed I think we can close this Issue.

**Comment by dandahle** [comment]
Thinking about this more the obvious finally hit me. The date and time entered is read and saved as GMT while the date and time retrieved and displayed is PDT (GMT -7 hours). Somewhere between the new entry and saving process there is a mismatch in time zones. Because the date and time doesn't change after refresh without any updates, it appears that once the date is saved the time zones are OK on the read or pull transaction to retrieve the date & time.

**Comment by rarafon** [comment]
Hi Guys,
Sorry for the delay. I noticed the bug when I made the changes for the month. It seems my local server is set up differently so I didn't have the error locally. I believe I should have corrected it in the recent #297 by using the entire string instead. I didn't know that JS could do that, but it should also be a better approach for the months as well since it doesn't converting the date anymore.

Hopefully, I didn't introduce any bugs.
